docs/ga-roadmap.md

```diff
@@ -0,0 +1,114 @@
+
+## Proposal
+
+This document proposes a GA Roadmap based on multiple phases, where different parts of the collector will be released as stable at different moments of time.
+
+At this moment we are defining only the first two phases of the process, and the next phases will be defined at a later stage once the Collector maintainers will have better understanding of the process and implications.
+
+When other signals data models (proto definition) will be marked as stable,
+
+### Phase 1
+
+**Tentative Date:** 2021-03-31
+
+**KR:** At the end of this phase the Collector’s core API/SPI will be marked as Stable.
+
+At the end of this phase we want to achieve core APIs stability. This will allow developers to implement internal components and extend the collector will be marked as stable. The complete list of the packages/modules will be finalized during the first action item of this phase, but the tentative list is:
+
+* `consumer`
+  * Official internal data model `pdata`.
+  * Interfaces and utils to build a Consumer for (trace, metrics, logs).
+* `config`
+  * Core `config` including service definition, internal components definition will be stabilized.
+  * To be determined which config helpers will be marked as stable (e.g. configgrpc, etc.).
+* `component`
+  * Interfaces and utils to build a Collector internal component (receiver, processor, exporter, extension).
+* `obsreport`
+  * Focus on the public API of this package. It is out of scope to ensure stability for the metrics emitted (focus in phase 2).
+* `service`
+  * Public API to construct a OpenTelemetry Collector Service.
+
+**Action Items:**
+
+* Create a new milestone for this phase, create issues for all the other action items and add them to the milestone.
+* Agreement on all packages/modules that will be marked as stable during this phase.
+* Write a version doc as per [version and stability document](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md).
+  * Previously it was discussed that for the Collector it is fine to release stable golang modules that contain APIs marked as experimental.
+  * Define status schema (experimental/stable), what are they applicable to every module.
+* Investigate if splitting into smaller, more granular, modules is possible.
+  * Define the modules schema, try to not break everyone. See [here](https://github.com/golang/go/wiki/Modules#is-it-possible-to-add-a-module-to-a-multi-module-repository).
+  * Investigate how can we release multiple golang modules from the same repo, without asking people that consume them to use a replace statement. See problem in contrib.
+  * Investigate how to release testonly utils.
+* Review all public APIs and godocs for modules that we want to release in this phase.
+  * Fix all critical issues, and remove unnecessary (“When in doubt leave it out”) public APIs.
```

Comment:
One more reason to have a dedicated spi package.

Reply:
Not sure what you understand by `SPI` :)

Reply:
Moving this discussion to the other comment about SPI :)